src/routers/handlers/enter_your_details/index.ts

```diff
@@ -56,24 +66,24 @@ export class EnterYourDetailsHandler extends GenericHandler<EnterYourDetailsView
 
     public executePost(req: Request, _response: Response): ViewModel<EnterYourDetailsViewData> | Redirect {
         logger.info(`${this.constructor.name} post execute called`);
-        const viewData = this.getViewData(req);
-        // generate the redirect query string
-        const redirect: string = `${PrefixedUrls.CHECK_DETAILS}`;
         // validating the form using the req object
         const errors = this.validateRequest(req);
 
         if (!errors.isEmpty()){
+            const viewData = this.getViewData(req);
             // if validation errors exists, get them as an array
             viewData.errors = this.convertValidationErrorsToErrorManifestType(errors.array());
             return {
                 templatePath: EnterYourDetailsHandler.templatePath,
                 viewData
             };
         }
-
-        const details =  generateUserDetails(req);
+        const address = isAddress(req.body) ? req.body : {} as Address;
+        const details =  generateUserDetails(req, address);
 
         setPresenterAccountDetails(req, details);
+        // generate the redirect query string
+        const redirect = PrefixedUrls.CHECK_DETAILS;
 
         return { redirect };
```

Comment:
As I stated in my previous review.
There is no need for this to be stored in a variable.
Just return an object literal with the redirect field set to `PrefixedUrls.CHECK_DETAILS`.